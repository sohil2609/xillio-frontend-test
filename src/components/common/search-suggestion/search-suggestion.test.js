import React from 'react';
import { shallow } from 'enzyme';
import SearchSuggestion from './search-suggestion.component';

const setUp = (props = {}) => {
	const component = shallow(<SearchSuggestion {...props} />);
	return component;
};

describe('Search Suggestion Component', () => {
	const mockProps = {
		list: [ 'abc', 'abcd', 'abcde' ],
		onClick: jest.fn(),
		noOfSuggestion: 2
	};
	const component = setUp(mockProps);

	it('Should test snapshot', () => {
		expect(component).toMatchSnapshot();
	});

	it('Should render only 2 suggestions', () => {
		const items = component.find('[data-test="searchItem"]');
		expect(items.length).toBe(2);
	});
});
