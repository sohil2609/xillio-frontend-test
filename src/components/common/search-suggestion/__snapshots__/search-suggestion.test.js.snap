// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo Item Component Should test snapshot 1`] = `ShallowWrapper {}`;

exports[`Search Suggestion Component Should test snapshot 1`] = `ShallowWrapper {}`;
